src/lib/signerVerifier.js

```diff
@@ -0,0 +1,76 @@
+const nacl = require('tweetnacl');
+const bs58 = require('bs58');
+const { TextEncoder } = require('util');
+const didUtil = require('./did');
+
+const textEncoder = new TextEncoder();
+
+class Ed25519Signer {
+  constructor(key, verificationMethod) {
+    this.key = key;
+    this.verificationMethod = verificationMethod;
+  }
+
+  sign(proof) {
+    const signed = nacl.sign.detached(textEncoder.encode(proof.merkleRoot), bs58.decode(this.key));
+    const signature = Buffer.from(signed).toString('hex');
+
+    return {
+      signature,
+      verificationMethod: this.verificationMethod,
+    };
+  }
+}
+
+/**
+ * Creates a signer from the provided information
+ *
+ * @param options Signer options:
+ * @param options.verificationMethod The verificationMethod for the signing key
+ * @param options.keypair The keypair to sign with
+ *    or
+ * @param options.privateKey The private key to sign with
+ *    or
+ * @param options.signer An object implementing a `sign(CvcMerkleProof)` method
+ */
+const signer = async (options) => {
+  if (!options.signer && !options.keypair && !options.privateKey) {
+    throw new Error('Either a signer, keypair or privateKey is required');
+  }
+
+  const { verificationMethod } = options;
+  let { signer: signerImpl } = options;
+
+  // Create a signer from keypair/key
+  if (!signerImpl) {
```

Comment:
consider an early return instead of this if block
```suggestion
  if (signerImpl) return signerImpl
```